python/cudf/cudf/core/column/column.py

```diff
@@ -2719,8 +2731,15 @@ def as_column(
             return as_column(arbitrary, dtype=dtype, nan_as_null=nan_as_null)
         elif arbitrary.dtype.kind in "biuf":
             from_pandas = nan_as_null is None or nan_as_null
-            return as_column(
+            if not is_np_native_byteorder(arbitrary.dtype):
```

Comment:
You can just use `dtype.isnative` here I believe, no need for a helper function: https://numpy.org/doc/2.2/reference/generated/numpy.dtype.isnative.html

```suggestion
            if not arbitrary.dtype.isnative:
```

Reply:
Ah great, didn't know about this API. Thanks!